nsqd: panic: runtime error: invalid memory address or nil pointer dereference

hi,
  when i use nsqd --statsd-address=... post data to statsd/Graphite  , the problem happend，my environment is :
```
Centos 6.5 64bit
golang version 1.7.4
nsqd version 1.0
```
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x51e003]

goroutine 38 [running]:
panic(0x7c1aa0, 0xc420010120)
        /usr/lib/golang/src/runtime/panic.go:500 +0x1a1
github.com/nsqio/nsq/nsqd.(*NSQD).statsdLoop(0xc42018c000)
        /data/go/src/github.com/nsqio/nsq/nsqd/statsd.go:126 +0x1983
github.com/nsqio/nsq/nsqd.(*NSQD).Main.func6()
        /data/go/src/github.com/nsqio/nsq/nsqd/nsqd.go:264 +0x2a
github.com/nsqio/nsq/internal/util.(*WaitGroupWrapper).Wrap.func1(0xc420161ba0, 0xc42018c0d8)
        /data/go/src/github.com/nsqio/nsq/internal/util/wait_group_wrapper.go:14 +0x27
created by github.com/nsqio/nsq/internal/util.(*WaitGroupWrapper).Wrap
        /data/go/src/github.com/nsqio/nsq/internal/util/wait_group_wrapper.go:16 +0x62
```

--- a/nsqd/http.go
+++ b/nsqd/http.go
@@ -506,11 +506,11 @@ func (s *httpServer) doStats(w http.ResponseWriter, req *http.Request, ps httpro
 		Health    string       `json:"health"`
 		StartTime int64        `json:"start_time"`
 		Topics    []TopicStats `json:"topics"`
-		Memory    *memStats    `json:"memory"`
+		Memory    memStats     `json:"memory"`
 	}{version.Binary, health, startTime.Unix(), stats, ms}, nil
 }
 
-func (s *httpServer) printStats(stats []TopicStats, ms *memStats, health string, startTime time.Time, uptime time.Duration) []byte {
+func (s *httpServer) printStats(stats []TopicStats, ms memStats, health string, startTime time.Time, uptime time.Duration) []byte {
 	var buf bytes.Buffer
 	w := &buf
 	now := time.Now()

--- a/nsqd/stats.go
+++ b/nsqd/stats.go
@@ -157,7 +157,7 @@ type memStats struct {
 	GCTotalRuns       uint32 `json:"gc_total_runs"`
 }
 
-func getMemStats() *memStats {
+func getMemStats() memStats {
 	var ms runtime.MemStats
 	runtime.ReadMemStats(&ms)
 
@@ -170,7 +170,7 @@ func getMemStats() *memStats {
 	copy(gcPauses, ms.PauseNs[:length])
 	sort.Sort(gcPauses)
 
-	return &memStats{
+	return memStats{
 		ms.HeapObjects,
 		ms.HeapIdle,
 		ms.HeapInuse,

--- a/nsqd/statsd.go
+++ b/nsqd/statsd.go
@@ -23,7 +23,7 @@ func (s Uint64Slice) Less(i, j int) bool {
 }
 
 func (n *NSQD) statsdLoop() {
-	var lastMemStats *memStats
+	var lastMemStats memStats
 	var lastStats []TopicStats
 	ticker := time.NewTicker(n.getOpts().StatsdInterval)
 	for {

